source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/ChargeLinkCommands/ChargeLinkCommandFactory.cs

```diff
@@ -40,6 +47,39 @@ public ChargeLinkCommandFactory(IChargeRepository chargeRepository, IClock clock
             string correlationId)
         {
             var charge = await _chargeRepository.GetChargeAsync(defaultChargeLink.ChargeId).ConfigureAwait(false);
+            return CreateChargeLinkCommand(
+                createLinkCommandEvent.MeteringPointId,
+                correlationId,
+                charge,
+                defaultChargeLink.GetStartDateTime(createLinkCommandEvent.StartDateTime),
+                defaultChargeLink.EndDateTime);
+        }
+
+        public async Task<ChargeLinkCommand> CreateFromChargeLinkAsync(
+            [NotNull] ChargeLink chargeLink,
+            [NotNull] ChargeLinkPeriodDetails chargeLinkPeriodDetails,
+            string correlationId)
+        {
+            var charge = await _chargeRepository.GetChargeAsync(chargeLink.ChargeId).ConfigureAwait(false);
+            var meteringPoint = await _meteringPointRepository
+                .GetMeteringPointAsync(chargeLink.MeteringPointId)
+                .ConfigureAwait(false);
+
+            return CreateChargeLinkCommand(
+                meteringPoint.MeteringPointId,
+                correlationId,
+                charge,
+                chargeLinkPeriodDetails.StartDateTime,
+                chargeLinkPeriodDetails.EndDateTime);
+        }
+
+        private ChargeLinkCommand CreateChargeLinkCommand(
+            string meteringPointId,
+            string correlationId,
+            Charge charge,
+            Instant startDateTime,
+            Instant? endDateTime)
```

Comment:
I think you can do this. When I look at it, both ways here will ensure end date is not null. Agree?

```suggestion
            Instant endDateTime)
```